### PR TITLE
SNAP-2071: Snappy UI becomes unresponsive on LowMemoryException

### PR DESCRIPTION
**Pull request by snappy-sachin**
## Changes proposed in this pull request

 - Adding timed wait while obtaining table stats from FunctionService in getStatsFromAllServers.
 - Return last successful updated table stats in case obtaining table stats takes more than 5 seconds or any exception is thrown.

## Patch testing

 - Verified manually and throwing LowMemoryException.
 - Verified using running job CreateAndLoadNWTablesSparkApp

## ReleaseNotes.txt changes

 - NA

## Other PRs 

 - NA

### CODE REVIEW

File: core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala [hunk 1]
```diff
@@ -164,18 +164,24 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
...
+    val currTableSizeInfo = tableSizeInfo.values.toSeq
     var result = new java.util.ArrayList[SnappyRegionStatsCollectorResult]().asScala
     var externalTables = scala.collection.mutable.Buffer.empty[SnappyExternalTableStats]
     val dataServers = GfxdMessage.getAllDataStores
     try {
       if (dataServers != null && dataServers.size() > 0) {
         result = FunctionService.onMembers(dataServers)
-            .withCollector(new GfxdListResultCollector())
-            .execute(SnappyRegionStatsCollectorFunction.ID).getResult().
+            // .withCollector(new GfxdListResultCollector())
```
Comment (rishitesh) [inline core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala]: Why is this commented? I don't know much about function execution, but was there any limitation?

File: core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala [hunk 2]
```diff
@@ -164,18 +164,24 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
 
   override def getStatsFromAllServers(sc: Option[SparkContext] = None): (Seq[SnappyRegionStats],
       Seq[SnappyIndexStats], Seq[SnappyExternalTableStats]) = {
+    val currTableSizeInfo = tableSizeInfo.values.toSeq
```
Comment (rishitesh) [inline core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala]: No need for a separate variable currTableSizeInfo.

File: core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala (hunk 1 above)
Comment (snappy-sachin) [reply core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala]: Yes, getResult with timeout is not expected to be invoked for GemFireXD.

File: core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala (hunk 2 above)
Comment (snappy-sachin) [reply core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala]: Removed unnecessary variable currTableSizeInfo.

**Comment by snappy-sachin** [comment]
Feedback is taken care and clarified the doubts.
Hence merging.

**Pull request merged**